src/inference/dev_api/ie_processor_type.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file for config that holds the processor type for CPU inference
+ * @file ie_processor_type.hpp
+ */
+
+#pragma once
+#include <ie_parameter.hpp>
+#include <ie_plugin_config.hpp>
+
+namespace InferenceEngine {
+
+struct ProcTypeConfig {
```

Comment:
why do we need this structure? we already have enumeration which can be used directly

Reply:
it is used for one implementation for conversion between property enumeration value and OV 1.0 string value. We use it for ie.GetConfig() and ie.SetConfig() in this PR. May use it for model.GetConfig() in the future.

Reply:
Let's examine the flow:
- User called Core::set_property
- Next it goes to InferencePlugin::SetConfig. Here enumeration value is converted to string automatically using Core default logic and operator `operator<<`
- Then plugin parses this value from string using `operator>>`

Looks like we can use new enumeration in current plugins even with Plugin API 1.0

Reply:
Yes, you are right. The new enumeration can be used with both API 1.0 and API 2.0 in this PR as the flow you mentioned. 

We also added key and value check for API 1.0 here.

Reply:
does it mean we can drop this structure?

Reply:
updated